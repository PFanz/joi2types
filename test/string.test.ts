import Joi from '@hapi/joi';
import Joi2Types, { joi2JsonSchema } from '../dist/index';

test('joi2Types string enum', async () => {
  const schema = Joi.string().valid('browser', 'hash', 'memory');
  expect(joi2JsonSchema(schema)).toEqual({
    type: 'string',
    enum: ['browser', 'hash', 'memory'],
  });

  const types = await Joi2Types(schema);
  expect(types).toMatchSnapshot();
});
