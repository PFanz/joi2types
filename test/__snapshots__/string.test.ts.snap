// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`joi2Types string enum 1`] = `
"export type JoiTypes = \\"browser\\" | \\"hash\\" | \\"memory\\";
"
`;
